beta/src/components/MDX/Sandpack/eslint-integration.tsx

```diff
@@ -0,0 +1,93 @@
+// @ts-nocheck
+
+import {Linter} from 'eslint/lib/linter/linter';
+
+import type {Diagnostic} from '@codemirror/lint';
+import type {Text} from '@codemirror/text';
+
+const getCodeMirrorPosition = (
+  doc: Text,
+  {line, column}: {line: number; column?: number}
+): number => {
+  return doc.line(line).from + (column ?? 0) - 1;
+};
+
+const linter = new Linter();
+
+// HACK! Eslint requires 'esquery' using `require`, but there's no commonjs interop.
+// because of this it tries to run `esquery.parse()`, while there's only `esquery.default.parse()`.
+// This hack places the functions in the right place.
+const esquery = require('esquery');
+esquery.parse = esquery.default?.parse;
+esquery.matches = esquery.default?.matches;
+
+const reactRules = require('eslint-plugin-react-hooks').rules;
+linter.defineRules({
+  'react-hooks/rules-of-hooks': reactRules['rules-of-hooks'],
+  'react-hooks/exhaustive-deps': reactRules['exhaustive-deps'],
+});
+
+const rules = linter.getRules();
+
+const options = {
+  parserOptions: {
+    ecmaVersion: 12,
+    sourceType: 'module',
+    ecmaFeatures: {jsx: true},
+  },
+  rules: {
+    'react-hooks/rules-of-hooks': 'error',
+    'react-hooks/exhaustive-deps': 'warn',
+    ...Array.from(rules).reduce((result, [ruleId, rule]) => {
```

Comment:
Do we really want to lint all kinds of errors?

Reply:
I've removed this since since we wanted only the two react hooks rules